test/server/cards/01_SOR/units/FirstLegionSnowTrooper.spec.js

```diff
@@ -0,0 +1,86 @@
+describe('First Legion Snow Trooper', function() {
+    integration(function() {
+        describe('First Legion Snow Trooper while attacking non-damaged unit.', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['first-legion-snowtrooper'],
+                    },
+                    player2: {
+                        groundArena: ['yoda#old-master'],
+                    }
+                });
+            });
+
+            it(' Ability should do nothing', function () {
+                this.player1.clickCard(this.firstLegionSnowtrooper);
+                this.player1.clickCard(this.yodaOldMaster);
+                expect(this.firstLegionSnowtrooper.exhausted).toBe(true);
+                expect(this.firstLegionSnowtrooper.damage).toBe(2);
+                expect(this.yodaOldMaster.damage).toBe(2);
+                expect(this.player2).toBeActivePlayer();
+            });
+        });
+        describe('First Legion Snow Trooper with a damaged enemy unit.', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [{ card: 'first-legion-snowtrooper' }],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'yoda#old-master', damage: 1 }],
+                        base: { card: 'dagobah-swamp', damage: 5 }
+                    }
+                });
+            });
+
+            it('First legion Snowtrooper should receive overwhelm and +2/+0, defeating yoda and dealing 1 damage to opponents base.', function () {
+                this.player1.clickCard(this.firstLegionSnowtrooper);
+                this.player1.clickCard(this.yoda);
+                expect(this.firstLegionSnowtrooper.exhausted).toBe(true);
+                expect(this.firstLegionSnowtrooper.damage).toBe(2);
+                expect(this.p2Base.damage).toBe(6);
+                expect(this.player2).toHaveEnabledPromptButtons(['You', 'Opponent', 'You and Opponent', 'No one']);
```

Comment:
Do you mind switching to a unit that doesn't have an on-defeat just to make the test as simple as possible

Reply:
When it came to this test i wanted to see how the conditional would react when Yoda was removed from play and the snowtrooper still had the isAttacking() True. If its unnecessary I'll change the unit.

Reply:
It's good to test the snow trooper isAttacking case, we don't have great test coverage for that yet. But we have really good tests for when-defeated ability timing, so we can omit that and switch to a unit that doesn't have one for simplicity

Reply:
Alrighty i'll swap to another unit.

Reply:
swapped to  SnowTrooper lieutenant